### PR TITLE
Change Make default target not to call `clean`

### PR DESCRIPTION
**Pull request by takanohi**
Picking this good first issue #1053 .

```
# Before
make == make all
make all == make binaries (with clean) manpages

# After
make == make binaries (without clean)
make all == make clean binaries manpages
```

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -75,7 +75,7 @@ jobs:
     - name: Unit tests
       run: go test -v ./...
     - name: Make
-      run: make
+      run: make all
```
Comment (AkihiroSuda) [inline .github/workflows/test.yml:78]: `make` has to continue to work

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
`make` w/o target has to continue to work

**Comment by afbjorklund** [comment]
It was also requested to continue to clean...

File: .github/workflows/test.yml (hunk 1 above)
Comment (takanohi) [reply .github/workflows/test.yml]: fixed

**Comment by afbjorklund** [comment]
I don't think this change is related to the issue that I opened, the question was if make could stop removing `vendor`

This is the code involved with that:

```Makefile
.PHONY: clean
clean:
	rm -rf _output vendor
```

If `make` (or `make all`, same thing) should call clean or not, is more of an optimization issue (over risk of stale build)...

**Review by afbjorklund — CHANGES_REQUESTED** [review]
I don't think we should add clean only to the "all" target, and remove it from `make`

**Comment by takanohi** [comment]
@afbjorklund 

You wrote a description in the issue.
`It is fine that it is removed on "clean", but annoying that it is deleted on "make".`

So changing default target will fix your problem immediately. (`make` doesn't remove vendor anymore.)
jandubois also had commented [here](https://github.com/lima-vm/lima/issues/1053#issuecomment-1250126025).

Do you want to remove `clean` process from the whole Makefile?
I think it will be our next action in another PR.

**Comment by afbjorklund** [comment]
I think they are two different topics, but I am also willing to just do nothing at all (leave as-is)...

If my vendor directory gets wiped, I can do `go mod vendor` again. And avoid to run `make all`
The running of clean and build in parallel is another topic, but there is a workaround in place.
And there are limitations to `go build`, so I think that what we have is "good enough" for now.

**Comment by afbjorklund** [comment]
@takanohi thanks for offering to help!

**Comment by takanohi** [comment]
If leaving as-is is ok for you, this PR can be closed.
It's helpful if you close the issue again.

**Pull request closed without merging**